examples/maturin-starter/src/lib.rs

```diff
@@ -19,7 +19,7 @@ impl ExampleClass {
 }
 
 /// An example module implemented in Rust using PyO3.
-#[pymodule]
+#[pymodule(supports_free_threaded = true)]
```

Comment:
I'm unsure of whether we should start applying this in our examples. As far as I know, free threaded support is more of a "it's experimental, you can play around with it if you want", not a "you should use this everywhere" thing. If our examples have it, people will copy it.

Reply:
We have some problems like https://github.com/PyO3/pyo3/pull/4544#issuecomment-2357062936 .

Reply:
I guess what I've said is the long version of "should be recommend that people use this?"

Reply:
After some reflection I think you're correct. Given that we intend to flip the default for this in 0.24 or 0.25 to opt-out once we've had some more time for it to bed in, I think it's premature to add this flag to examples for now.

Sure, users will end up writing extensions which accidentally force the GIL on a freethreaded build. But I think that's a better choice than throwing freethreading at users who might themselves be just learning Rust and/or Python for the first time.

Once we flip the default, things will likely just work for them without them having to care.

Reply:
Are you OK leaving the flag in the raw FFI examples? I think it's helpful to show exactly how to do it there.

Reply:
I think the raw FFI example makes sense to have it, yes. Our default won't apply there, and users doing raw FFI are:
- presumably not beginners 👍 
- have chosen to do it the hard way already 😂